asapdiscovery-docking/asapdiscovery/docking/scripts/run_docking_oe.py

```diff
@@ -200,6 +157,85 @@ def mp_func(out_dir, lig_name, du_name, *args, **kwargs):
     return results
 
 
+def parse_du_filenames(receptors, regex, basefile="predocked.oedu"):
+    """
+    Parse list of DesignUnit filenames and extract identifiers using the given regex.
+    `regex` should have one capturing group (which can be the entire string if desired).
+
+    Parameters
+    ----------
+    receptors : Union[List[str], str]
+        Either list of DesignUnit filenames, or glob/directory/file to load from. If a
+        file is passed, will assume this is a CSV file and will load from the
+        "Docked_File" column using `basefile`
+    regex : str
+        Regex string for parsing
+    basefile : str, default="predocked.oedu"
+        If a CSV file is passed for `receptors`, this is the base filename that will be
+        appended to every directory found in the "Docked_File" column
+
+    Returns
+    -------
+    Dict[str, List[str]]
+        Dictionary mapping Mpro dataset name/compound id to list of full
+        Mpro names/compound ids (with chain)
+    Dict[str, str]
+        Dictionary mapping full name (with chain) to full filename
+    """
+    from asapdiscovery.data.utils import construct_regex_function
+
+    # First get full list of filenames
+    if type(receptors) is list:
+        print("Using files as given")
+        all_fns = receptors
+    elif os.path.isdir(receptors):
+        print(f"Using {receptors} as directory")
+        all_fns = [
+            os.path.join(receptors, fn)
+            for _, _, files in os.walk(receptors)
+            for fn in files
+            if fn[-4:] == "oedu"
+        ]
+    elif os.path.isfile(receptors):
+        print(f"Using {receptors} as file")
+        df = pandas.read_csv(receptors)
+        all_fns = [
+            os.path.join(os.path.dirname(fn), basefile) for fn in df["Docked_File"]
```

Comment:
Perhaps a `try: except` to check if there is a col called `Docked_File` and then a prettier error message than a `KeyError`?